gateway-ha/src/main/java/io/trino/gateway/ha/security/LbLdapClient.java

```diff
@@ -109,15 +115,18 @@ public String getMemberOf(String user) {
     String filter = config.getLdapUserSearch().replace("${USER}", user);
 
     String[] attributes = new String[]{config.getLdapGroupMemberAttribute()};
-    List<UserRecord> list = ldapConnectionTemplate.search(config.getLdapUserBaseDn(),
+    List<Optional<UserRecord>> list = ldapConnectionTemplate.search(config.getLdapUserBaseDn(),
         filter,
         SearchScope.SUBTREE,
         attributes,
         userRecordEntryMapper);
 
     String memberOf = "";
     if (list != null && !list.isEmpty()) {
-      memberOf = list.listIterator().next().getMemberOf();
+      Optional<UserRecord> userRecord = list.listIterator().next();
```

Comment:
Is it possible that some items in the list might have the `memberOf` attribute defined an others won't? We just take the first item in the list here, so it looks like we assume that that everything returned by the search will be the same with respect to the memberOf attribute. If this isn't the case, you could filter the list using sth like `stream().filter(e -> e.isPresent())`.

Reply:
good point. I don't see anywhere where we validate that the configured `filter` returns a unique result though. I assume it is meant to be something like `sAMAccountName=${USER}`. Maybe I have missed that though, @vishalya could you chime in here?

Reply:
@vishalya any thoughts?

Reply:
memberOf should be a single record of string type, at least that's what we have here. The string is `,` delimited list.
You change looks generic, but we will have to use config to hint at what field to use in order to find the membership.

Reply:
`ldapGroupMemberAttribute` should help the configuration and might void this issue.

Reply:
Can you elaborate when you mean by "use config to hint at what field to use in order to find the membership."

Also @willmostly , if I run the test codes and it fails on [line 126](https://github.com/trinodb/trino-gateway/pull/31/files#diff-d59eac35750dcc974b98c0c5387a7ae147e3a3fbb339b4c1c5d2b4a1f39bb8a1R126). It seems that the [map function](https://github.com/trinodb/trino-gateway/pull/31/files#diff-d59eac35750dcc974b98c0c5387a7ae147e3a3fbb339b4c1c5d2b4a1f39bb8a1R47-R51) that is overriding has issues when changed to Optional. Do you happen to know how to fix this? 
```
java.lang.ClassCastException: class io.trino.gateway.ha.security.LbLdapClient$UserRecord cannot be cast to class java.util.Optional (io.trino.gateway.ha.security.LbLdapClient$UserRecord is in unnamed module of loader 'app'; java.util.Optional is in module java.base of loader 'bootstrap')
```